### DISCUSSION TITLE
How to enable RNN?

### DISCUSSION THREAD

**Discussion by zxzzz0** [opendilab/DI-engine#105] [Q&A] [answered]
Found [this file](https://github.com/opendilab/DI-engine/blob/main/ding/torch_utils/network/rnn.py) but not sure how to change [this config](dizoo/slime_volley/config/slime_volley_ppo_config.py) to enable RNN.

**Comment by PaParaZz1** [comment, marked as answer]
You can refer to this [best practice](https://opendilab.github.io/DI-engine/best_practice/rnn.html).

**Reply by zxzzz0** [reply]
Could we know why PPO doesn't have RNN support?

**Reply by PaParaZz1** [reply]
We think it is easy to add RNN in `PPOPolicy` following our doc. And RNN does not always lead to performance gain due to training instability and hyper-parameter sensetity, so we don't add RNN in every policy by default. If you think it is necessary, we will add a `PPORNNPolicy` as soon as possible.

**Reply by zxzzz0** [reply]
Understood. You don't need to spend time on this. I can try myself. Thanks!
